NuRadioReco/modules/channelMeasuredNoiseAdder.py

```diff
@@ -0,0 +1,142 @@
+        self.__noise_data = None
+
+    def begin(self, filenames, random_seed=None, max_iterations=100, debug=False, draw_noise_statistics=False):
+        """
+        Set up module parameters
+
+        Parameters:
+        ------------------
+        filenames: list of strings
+            List of .nur files containing the measured noise
+        random_seed: int, default: None
+            Seed for the random number generator. By default, no seed is set.
+        max_iterations: int, default: 100
+            The module will pick a random event from the noise files, until a suitable event is found
+            or until the number of iterations exceeds max_iterations. In that case, an error is thrown.
+        debug: bool, default: False
+            Set True to get debug output
+        draw_noise_statistics: boolean, default: False
+            If true, the values of all samples is stored and a histogram with noise statistics is drawn
+            be the end() method
+        """
+        self.__filenames = filenames
+        self.__io = NuRadioReco.modules.io.NuRadioRecoio.NuRadioRecoio(self.__filenames)
+        self.__random_state = numpy.random.RandomState(random_seed)
+        self.__max_iterations = max_iterations
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
+            self.logger.debug('Reading noise from {} files containing {} events'.format(len(filenames), self.__io.get_n_events()))
+        if draw_noise_statistics:
+            self.__noise_data = []
+
+    @register_run()
+    def run(self, event, station, det):
+        """
+        Add measured noise to station channels
+
+        Parameters:
+        --------------
+        event: event object
+        station: station object
+        det: detector description
+        """
+        noise_station = None
+        for i in range(self.__max_iterations):
+            noise_station = self.get_noise_station(station)
+            # Get random station from noise file. If we got a suitable station, we continue,
+            # otherwise we try again
+            if noise_station is not None:
+                break
+        # To avoid infinite loops, if no suitable noise station was found after a number of iterations we raise an error
+        if noise_station is None:
+            raise ValueError('Could not find suitable noise event in noise files after {} iterations'.format(self.__max_iterations))
+        for channel in station.iter_channels():
+            noise_channel = noise_station.get_channel(channel.get_id())
+            channel_trace = channel.get_trace()
+            if noise_channel.get_sampling_rate() != channel.get_sampling_rate():
+                noise_channel.resample(channel.get_sampling_rate())
+            noise_trace = noise_channel.get_trace()
+            if channel.get_number_of_samples() > noise_channel.get_number_of_samples():
+                self.logger.warning(
```

Comment:
is a warning enough? Depending on how long the simulated trace is, this might lead to wrong results.

Reply:
I changed the get_noise_station method to only accept stations if the noise trace is at least as long as the data trace.